src/main/java/com/xceptance/xlt/engine/DataManagerImpl.java

```diff
@@ -70,6 +72,8 @@ public class DataManagerImpl implements DataManager
      * Logger responsible for logging the statistics to the timer file(s).
      */
     private volatile BufferedWriter logger;
+    
+    private HashMap<String, DataLogger> dataLoggers = new HashMap<String, DataLogger>();
```

Comment:
I would feel better if this map would be a `ConcurrentHashMap`. In the rare case we have parallel activities in a single session.

Furthermore, please add some javadoc.